core/interop/IDynamicInterfaceCastable/README.md

```diff
@@ -0,0 +1,75 @@
+---
+languages:
+- csharp
+- cpp
+products:
+- dotnet-core
+page_type: sample
+name: "IDynamicInterfaceCastable: Supporting Interfaces Dynamically"
+urlFragment: "idynamicinterfacecastable"
+description: "A .NET application that shows how to implement IDynamicInterfaceCastable to project a native object as implementing different managed interfaces."
+---
+
+# `IDynamicInterfaceCastable` Sample
+
+The [`IDynamicInterfaceCastable` API](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.idynamicinterfacecastable) was introduced in .NET 5 as a way for creating a .NET class that supports interfaces which are not in its metadata.
+
+This sample provides an implementation of `IDynamicInterfaceCastable` that projects a native object as implementing different known managed interfaces. It uses COM conventions (e.g. `QueryInterface`) for interacting with the native object, but does not require the actual COM system.
+
+Note: The sample uses `Marshal` APIs as part of interacting with the native library. The introduction of [C# function pointers](https://github.com/dotnet/csharplang/blob/994c41586e07e38fb6b30902b1715b4025d80c52/proposals/function-pointers.md) will allow that interaction to occur in a more performant manner.
+
+## Prerequisites
+
+* [.NET Core 5.0 SDK](https://dotnet.microsoft.com/download) Preview 7 or later
+
+* C++ compiler
+  * Windows: `cl.exe`
+    * See [installation instructions](https://docs.microsoft.com/cpp/build/building-on-the-command-line#download-and-install-the-tools).
+  * Linux/OSX: `g++`
+
+## Build and Run
+
+1) In order to build and run, all prerequisites must be installed. The following are also required:
+
+    * On Linux/OSX, the C++ compiler (`g++`) must be on the path.
+    * The C++ compiler (`cl.exe` or `g++`) and `dotnet` must be the same bitness (32-bit versus 64-bit).
+      * On Windows, the sample is set up to use the bitness of `dotnet` to find the corresponding `cl.exe`
+
+1) Navigate to the root directory and run `dotnet build`
```

Comment:
Is this step extra and unneeded? The step after it already does `dotnet run` or `dotnet build`.

Reply:
```suggestion
1) Navigate to the root directory.
```